requirements-unlocked.txt

```diff
@@ -0,0 +1,3 @@
+httpx
```

Comment:
**Top-level requirements**
@chrisroederucdenver For my Python projects that are not using `tox` or `poetry`, I treat `requirements.txt` as the lockfile and create a file for top level requirements called `requirements-unlocked.txt`. I don't usually put versions unless I run into issues, in which case find that locking it to a major version number is adequate (and if not, usually a minor version number works).

I _think_ these 3 items are adequate for the top level requirements, but not gonna spend too much time on this right now as I usually don't run into any problems. But if we do, we can fine tune further.

Reply:
@chrisroederucdenver Copy/pasting your comments here:
> I haven't used requirements-unlocked.txt. It looks good in principle. Is it just as good to have requirements.txt just > have what you have here for requirements-unlocked.txt??
> I just ran freeze >> requirements.txt out of expediency. I'm not married to what must be a list of junk.

Looks like we're on the same page w/ this one.

Also, one practice I like is  to try to keep as many comments as possible 'review comments'. That is, the kind of comments that open these threads where you can mark "Resolve conversation" to hide them when done. I find it helps really reduce the clutter. You may already know how to start such a thread, but if you don't, you can start a new one by selecting the "Files change" tab, then clicking the + sign next to a line of code, adding your comment, and then clicking the appropriate button at the top right of the screen. 

That being said, when we come to an agreement / if you approve of the changes in the thread, feel free to click "Resolve conversation".

Reply:
(all confused because lockfile means something different in primitive concurrency solutions from long ago) I get it, lock, like locking the revisions of the dependencies. I get it. Thanks. And yeah, when someone gets around to creating a new venv, we'll know if you got it right. I agree, nothing to get excited about.

re: less clutter. This. way. I'll look for stuff. Feeling a bit swamped this week.